### ISSUE TITLE
Hot to customize traits

### ISSUE THREAD

**Issue by narraressan** [artf/grapesjs#65]
Hi. Thank you for helping me out in my previous questions. But here I am again...

I currently have this setup

grapesjs.init({
   ...
   panels: {
      defaults: [
         {
	     id: 'views',
	     buttons: [
                 ... {
			id: 'open-trait-manager',
			className: 'fa fa-cog',
			command: ''
		}...]
	 } 
      ]
   }
   traitManager: {
       labelContainer: "Component settings"
   }...
});

I am not sure of what I am doing here but its definitely not correct and I would like to customize the components settings (I believe, those are the 'traits') right?

**Comment by artf** [comment]
Hi @narraressan, yeah, component settings are what I call 'traits'. 
Anyway, did you already checked the wiki [here](https://github.com/artf/grapesjs/wiki/Components#define-new-component)? Traits are relative to the component so you define them on component level.
I'm still working on the wiki about Traits so if you have any question or just tell what you'd like to achieve maybe I can help you here.

**Comment by narraressan** [comment]
Hi, thanks for the information. This is exactly what I wanted, to change the type and id of the input elements...

But how do I activate the view for the trait in the editor, I tried to copy and paste the example but I can't find the way to activate the trails editor, I am thinking it should be something similar with opening the block manager in the panel 

panels: { 
  ...
  defaults: [{
    id: 'open-block-manager',
    className: 'fa fa-edit',
    command: 'open-blocks',
    active: true
  }]
  ...
}

but how do i do this for the traits?

**Comment by artf** [comment]
yeah sure:

```js
panels: { 
  ...
  defaults: [{
    id: 'open-tm',
    className: 'fa fa-cog',
    command: 'open-tm',
  }]
  ...
}
```

**Comment by NicholasGati** [comment]
Hi @artf, you're currently helping me with another issue but since I found this thread and I have a question that is in regards to traits and their relationship to components, I figured that this thread would be the more appropriate place to ask. Is it possible for a trait of "Component A" to change, update or affect the rendered canvas element for "Component B"? To explain it more, I have a list, which we'll call "Component A" and a list item, "Component B". When adding "Component B" to "Component A", I want to append a text input field in the component settings for "Component A" so that I can write into that text input field and have it update "Component B" text in the rendered list item inside of the canvas. So, I don't want to have the input field in the component settings of "Component B", even though it's updating that component. I want it in the component settings of "Component A". Is this possible? (Apologies if my explanation and question are not clear.)
